src/pushsource/_impl/backend/staged/staged_cloud.py

```diff
@@ -0,0 +1,136 @@
+        name = image.get("path")
+        src = os.path.join(origin, name)
+        build_info = KojiBuildInfo(
+            name=build_resources.get("name"),
+            version=build_resources.get("version"),
+            release=build_resources.get("respin"),
+        )
+        image_kwargs = {
+            "name": name,
+            "src": src,
+            "build_info": build_info,
+            "origin": origin,
+            "dest": [dest],
+            "sha256sum": image.get("sha256sum"),
+        }
+
+        image_kwargs.update(
+            {
+                "boot_mode": (
+                    BootMode(resources.get("boot_mode"))
+                    if resources.get("boot_mode")
+                    else None
+                )
+            }
+        )
+
+        release_required = ["product", "date", "arch", "respin"]
+        if all(x in release_resources.keys() for x in release_required):
+            release_attrs = [
+                "product",
+                "date",
+                "arch",
+                "respin",
+                "version",
+                "base_product",
+                "base_version",
+                "variant",
+                "type",
+            ]
+            release_kwargs = {}
+            for key in release_attrs:
+                release_kwargs[key] = release_resources.get(key)
+
+            image_kwargs["release"] = AmiRelease(**release_kwargs)
+
+        image_attrs = [
+            "type",
+            "region",
+            "virtualization",
+            "volume",
+            "root_device",
+            "description",
+            "sriov_net_support",
+            "ena_support",
+            "uefi_support",
+            "public_image",
+            "release_notes",
+            "usage_instructions",
+            "recommended_instance_type",
+            "marketplace_entity_type",
+            "image_id",
+            "scanning_port",
+            "user_name",
+            "version_title",
+            "security_groups",
+            "access_endpoint_url",
+        ]
+
+        for key in image_attrs:
+            if key in resources:
+                image_kwargs[key] = resources.get(key)
+
+        return AmiPushItem(**image_kwargs)
+
+    def __build_azure_push_item(self, resources, origin, image, dest):
+        build_resources = resources.get("build")
+        name = image.get("path")
+        src = os.path.join(origin, name)
+        build_info = KojiBuildInfo(
+            name=build_resources.get("name"),
+            version=build_resources.get("version"),
+            release=build_resources.get("respin"),
+        )
+        image_kwargs = {
+            "name": name,
+            "src": src,
+            "description": resources.get("description"),
+            "build_info": build_info,
+            "origin": origin,
+            "dest": [dest],
+            "sha256sum": image.get("sha256sum"),
+        }
+        return VHDPushItem(**image_kwargs)
```

Comment:
Sorry I didn't notice this before but it seems you're not creating the `VMIRelease` for `VHDPushItem`, this might break the publishing as you can [see here](https://github.com/release-engineering/pubtools-marketplacesvm/blob/main/src/pubtools/_marketplacesvm/cloud_providers/ms_azure.py#L105-L144) and [here](https://github.com/release-engineering/pubtools-marketplacesvm/blob/main/src/pubtools/_marketplacesvm/cloud_providers/ms_azure.py#L305)

Reply:
Oh, we will need to add date and product then.

Reply:
tyvm